frontend/gateway/grpcclient/client.go

```diff
@@ -824,12 +826,20 @@ func (ctr *container) Start(ctx context.Context, req client.StartRequest) (clien
 				if exit.Code == 0 {
 					continue
 				}
-				exitError = errors.Wrap(
-					&errdefs.ExitError{
-						ExitCode: exit.Code,
-					},
-					exit.Error,
-				)
+				if exit.Code == containerd.UnknownExitStatus {
```

Comment:
Don't quite understand this special case handling. Why isn't this just

```
err = grpcerrors.FromGRPC(status.ErrorProto(exit.Error)
if exit.Code != containerd.UnknownExitStatus {
   err = &errdefs.ExitError{ExitCode: exit.Code, Err: err}
}
```

Reply:
Yeah, that makes more sense.  `exit.Error` is an `*rpc.Status` so has to be converted to `*spb.Status` first for `status.ErrorProto`, unless I am missing some other helper func that already does this?  Anyway, updated per your outline here: https://github.com/moby/buildkit/pull/1627/commits/398593b61d9e77dbb4fc5597c567e436f7a9123c